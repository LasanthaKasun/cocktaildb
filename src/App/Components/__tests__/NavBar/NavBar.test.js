import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { NavBar } from "../..";

it("should render button component", () => {
  const component = renderer.create(<NavBar />);
  expect(component).toMatchSnapshot();
});

it("should render button with props", () => {
  render(<NavBar title="Sample Title"/>);
  const navBar = screen.getByTestId("navBar");
  expect(navBar).toHaveTextContent("Sample Title");
});
