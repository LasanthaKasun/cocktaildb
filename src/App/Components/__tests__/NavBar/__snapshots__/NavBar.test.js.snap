// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button component 1`] = `
<div
  className="fixed w-full z-30 flex bg-white dark:bg-[#0F172A] p-2 items-center justify-center h-16 px-10"
  data-testid="navBar"
>
  <div
    className="logo ml-12 dark:text-white  transform ease-in-out duration-500 flex-none h-full flex items-center justify-center"
  />
  <div
    className="grow h-full flex items-center justify-center"
  />
  <div
    className="flex-none h-full text-center flex items-center justify-center"
  >
    <div
      className="flex space-x-3 items-center px-3"
    >
      <div
        className="flex-none flex justify-center"
      >
        <div
          className="w-8 h-8 flex "
        >
          <img
            alt="profile"
            className="shadow rounded-full object-cover"
            src="logo.jpg"
          />
        </div>
      </div>
      <div
        className="hidden md:block text-sm md:text-md text-black dark:text-white"
      >
        Hi Amused
      </div>
    </div>
  </div>
</div>
`;
